### ISSUE TITLE
Flickering on 0.15.0-b1

### ISSUE THREAD

**Issue by Plawasan** [Aircoookie/WLED#3837]
### What happened?

After updating to the latest beta (0.15.0-b1) I see intermittent flickering on my 2D matrix clock. Downgrade to 0.14.2 resolves the issue.

### To Reproduce Bug

Upgrade to latest beta, create a 2D segment (8x32), use the scrolling text preset with #HHMM as segment name

### Expected Behavior

No flicker

### Install Method

Binary from WLED.me

### What version of WLED?

0.15.0-b1

### Which microcontroller/board are you seeing the problem on?

ESP32

### Anything else?

Runnning on ESP32

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Plawasan** [comment]
https://github.com/Aircoookie/WLED/assets/73796046/95008b06-008e-4fbc-a2bc-3e597329f699

**Comment by blazoncek** [comment]
Please try erasing flash. Sometimes it helps.

**Comment by botflakes** [comment]
> Please try erasing flash. Sometimes it helps.

Tried that, didn't solve the issue. Seems to be a bug within 0.15.0-b1. Rolling back to a stable release fixes this

**Comment by blazoncek** [comment]
Please try increasing flash speed from (default) 40MHz to 80MHz.
You will need a special build or manual parameter in esptool.
Be warned that not all flash chips can support 80MHz.

BTW I am assuming you have level shifters in place.

**Comment by Plawasan** [comment]
I need to crack the clock open to get to the ESP and flash it but in the meantime - no, I do not have a level shifter in place (the data lead is about 10cm long) however same as @botflakes - downgrading to 0.14.2 fixes the issue reliably

**Comment by blazoncek** [comment]
Nevertheless, the quality of the signal very much depends on the quality of the on-board PSU. 0.15. will consume more power than 0.14 and can introduce output level fluctuations - hence the recommendation for levelshifter.
You can also try adding 10uF capacitor (or 1uF) to 3.3V line.

**Comment by Plawasan** [comment]
>  You can also try adding 10uF capacitor (or 1uF) to 3.3V line.

That I have... I will also try to put the latest build on an Athom controller and see if that makes any difference compared to my DIY ESP32 board

**Comment by blazoncek** [comment]
The other possibility is CPU starvation. CPU may not be able to cope with load and skip a strict timing needed for LEDs.
I can see that on my test device with 6 usermods and 5 different outputs enabled. It flashes (occasionally) when reading temperature from Dallas sensor.

**Comment by blazoncek** [comment]
Try beta 2 posted on Discord and report.

**Comment by Plawasan** [comment]
> Try beta 2 posted on Discord and report.

first 10 minutes look good - not a single flicker. Will keep an eye on it.

**Comment by Plawasan** [comment]
No issues for more than a week - closing as resolved (for me at least).

**Comment by AStoker** [comment]
Hate to raise this one from the dead, but seems related. I'm running 0.15.3 (using a DigQuad with the ESP32 from QuinLED) and using 12v WS2811 bullet style led pixels. I have 310 LEDs forming 5 arches of 62 LEDs/arch. I'm using 3 data lines to drive the arches. So I start with injecting data and power on arch 1, which then links up to arch 2. Then I have another data line and power driving arch 3 and 4, and then a final data line and power driving arch 5.

When I use these all as a single 1d strip, everything is great, no flickering, just perfection. However, when I set up the 2D matrix, that's when I start getting weird flashing at about 3Hz. The flashing is very consistent, and doesn't change with effect speed, brightness, or anything else. I used an oscilloscope to look at the data line, and to map the power driving the LEDs, and everything looked fine (no voltage drop, no weird patterns).

I then noticed my last arch was not in fact flickering like the rest. I tried swapping out cables to ensure it wasn't a hardware issue, and verified that all the cables work as expected. So I then went to the software and configured my first data line (which drives two arches of 124 LEDs) to be the same last the last and only drive 62 LEDs (one arch). The blinking went away!

So, something weird is happening with the 2D matrix setup where I can't drive more than 62 (it's probably not exactly 62, but some value around there) LEDS without flashing starting to happen.

 I tried rolling back to 0.15.2, but it failed there too. I could keep going back to identify when the issue came up, but wanted to post here to get some help (and inform anybody else dealing with this that they're not crazy).
